File: problems/1796-second-largest-digit-in-a-string/solution.java
class Solution {
    public int secondHighest(String s) {
        int largest = -1;
        int secondLargest = -1;

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (Character.isDigit(c)) {
                int digit = c - '0';
                if (digit > largest) {
                    secondLargest = largest;
                    largest = digit;
                } else if (digit < largest && digit > secondLargest) {
                    secondLargest = digit;
                }
            }
        }

        return secondLargest;
    }
}

File: problems/1796-second-largest-digit-in-a-string/solution.py
class Solution:
    def secondHighest(self, s: str) -> int:
        digits = set()
        for char in s:
            if char.isdigit():
                digits.add(int(char))
        
        if len(digits) < 2:
            return -1
        
        digits = sorted(list(digits), reverse=True)
        return digits[1]